### PR TITLE
mgr/cephadm: refresh ceph.conf when mons change

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -2265,10 +2262,26 @@ def _apply_all_services(self):
...
+        last_monmap: Optional[datetime.datetime] = datetime.datetime.strptime(
+            monmap['modified'], CEPH_DATEFMT)
+        if last_monmap and last_monmap > datetime.datetime.utcnow():
+            last_monmap = None   # just in case clocks are skewed
+
         daemons = self.cache.get_daemons()
         grafanas = []  # type: List[orchestrator.DaemonDescription]
         for dd in daemons:
+            # orphan?
+            if dd.service_name() not in self.spec_store.specs and \
```
Comment (jschmid1) [inline src/pybind/mgr/cephadm/module.py]: In a post upgrade scenario, this would remove services, right?

Comment (liewegas) [reply src/pybind/mgr/cephadm/module.py]: yes.  if you have a cluster running octopus dev code, make sure you create the specs before upgrading.

Comment (jschmid1) [reply src/pybind/mgr/cephadm/module.py]: This is probably something we can automate in a nautilus->octopus upgrade.

It shouldn't be terribly hard to translate running services to valid ServiceSpecs.

Comment (liewegas) [reply src/pybind/mgr/cephadm/module.py]: no such thing as a nautilus->octopus upgrade with cephadm...

we do need to adjust the adoption workflow, though.  i'll update the docs.

**Comment by sebastian-philipp** [comment]
partially fixes https://tracker.ceph.com/issues/43705

File: src/pybind/mgr/cephadm/module.py (hunk 1 above)
Comment (sebastian-philipp) [reply src/pybind/mgr/cephadm/module.py]: well, there is going to be a deepsea to cephadm workflow needed for octopus

**Pull request merged**